src/components/SamlConfiguration/index.jsx

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Collapsible, Icon } from '@edx/paragon';
+import SamlProviderConfigForm from './SamlProviderConfigForm';
+import SamlProviderDataForm from './SamlProviderDataForm';
+import { camelCaseObject, snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import LoadingMessage from '../LoadingMessage';
+import ErrorPage from '../ErrorPage';
+
+class SamlConfiguration extends React.Component {
+  state = {
+    providerConfig: undefined,
+    providerData: undefined,
+    error: undefined,
+    loading: true,
+    deleteEnabled: false,
+  };
+
+  componentDidMount() {
+    LmsApiService.getProviderConfig(this.props.enterpriseId)
+      .then((response) => {
+        this.setState({
+          providerConfig: response.data.results[0],
+          loading: false,
+        });
+      })
+      .catch(error => this.setState({
+        error,
+      }));
+    LmsApiService.getProviderData(this.props.enterpriseId)
+      .then((response) => {
+        this.setState({
+          providerData: response.data.results[0],
+          loading: false,
+        });
+      })
+      .catch(error => this.setState({
+        error,
+        loading: false,
+      }));
+  }
+  /**
+   * Creates a new third party provider configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createProviderConfig = async (formData) => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('name', this.props.enterpriseName);
+    transformedData.append('slug', this.props.enterpriseSlug);
+    transformedData.append('enterprise_customer_uuid', this.props.enterpriseId);
+    try {
+      const response = await LmsApiService.postNewProviderConfig(transformedData);
+      this.setState({ providerConfig: response.data });
+      return undefined;
+    } catch (error) {
+      return error;
```

Comment:
Should we also call [`logError`](https://github.com/edx/frontend-app-admin-portal/blob/9fe269b344b1237d4b2a0e04201bbc24aad7091d/src/data/services/NewRelicService.js#L8) so the error is tracked in NewRelic? 

Also applies to the few other places where we catch/return an error.

Reply:
hmmm didn't know about this method.
Two questions:
1. It seems like I should be either handling the error format here or have the `logError` call in my child components that get the returned error. Is there a preferred place to put that? It seems like it would be best for most of the logic to live here. rather than in the child, but just want to be sure.
The reason I mention the error formats is because if the server does fail with a 500, the error data is ridiculously large. Django spits out a lot of unnecessary data (this is also why I use `error.message` in the case of 500s)

2. Do we want errors that are going to be seen by users to also be tracked? (It's absolutely fine either way, but in the past I've seen distinguishing from backend and frontend errors usually just either not tracked or tracked in different systems, so just confirming how we do things around here. :) )

Reply:
Ah also in trying to use the logError wrapper I noticed that it seems we've not been using the logError method at least not from the other edx libraries that have defined it (such as frontend-platform/logging).

It looks like in here specifically every subcomponent of the portal is being wrapped by  this ErrorBoundary class, which has: 
```
class ErrorBoundary extends React.Component {
  componentDidCatch(error) {
    NewRelicService.logError(error);
  }
```
I see the guide on Error Boundaries suggests try/catch doesn't actually use ErrorBoundaries, but I am unsure about the then/catch flow we're using.

Reply:
If an API call is wrapped with a try/catch, you're catching the exception before it makes it's way to the ErrorBoundary. The ErrorBoundary is to catch any exceptions/errors that are not explicitly caught (e.g., in a try/catch).

I think actually should have mentioned it's the `logAPIErrorResponse` function, not `logError`. I'd say modify either of those functions as you see fit for your use case.

For context, this repo is using an outdated implementation. There is a ticket on the backlog to refactor to use `@edx/frontend-build` and `@edx/frontend-platform` like newer MFEs use today, which would give us the ability to use `logError` from `@edx/frontend-platform/logging`.

The migration to use those ^^ packages may be non-trivial so we haven't tackled that quite yet.